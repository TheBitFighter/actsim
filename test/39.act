deftype struct (int a, b) { }

function foo (int x) : struct
{ 
   struct p;
   chp {
     p.a := x;
     p.b := x+1;
     self := p
   }
}


defproc test()
{
  struct w;

  chp {
    w := foo(4);
    log ("got: ", w.a, " and ", w.b)
  }
}

test t;
