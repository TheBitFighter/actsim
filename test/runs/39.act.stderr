WARNING: test<>: substituting chp model (requested prs, not found)
FATAL: What type is this (9) in ~ChpSimGraph()?
